### PR TITLE
ExternalLink: Convert component to TypeScript

### PR DESCRIPTION
**Pull request by walbo**
## What?
Converts the `ExternalLink` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
Refactors the current `ExternalLink` to TypeScript.

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Fire up the Storybook and confirm the `ExternalLink` continues to function as expected
3. Test the various external links within the block/site editors in Gutenberg still work

### CODE REVIEW

File: packages/components/src/external-link/types.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+export type ExternalLinkProps = {
+	/**
+	 * The content to be displayed within the link.
+	 */
+	children?: ReactNode;
```
Comment (mirka) [inline packages/components/src/external-link/types.ts]: I feel like we should mark `children` as a required prop, no?

I'm also wondering if we should explicitly mark `href` as a required prop. What do you think? (cc @ciampo)

**Review by mirka — COMMENTED** [review]
Looks good to me! I just had a few questions about explicitly marking required props.

File: packages/components/src/external-link/types.ts (hunk 1 above)
Comment (walbo) [reply packages/components/src/external-link/types.ts]: 👍 I agree. Will update later today if no objections.

Comment (ciampo) [reply packages/components/src/external-link/types.ts]: Agreed on marking both `children` and `href` as required!

We should update the README accordingly too

**Review by ciampo — APPROVED** [review]
LGTM 🚀

File: packages/components/src/external-link/types.ts (hunk 1 above)
Comment (walbo) [reply packages/components/src/external-link/types.ts]: 👍  Have marked them as required and synced the README file

**Pull request merged**